js/README.md

```diff
@@ -0,0 +1,301 @@
+[![Sandbox](https://img.shields.io/badge/status-sandbox-red?style=for-the-badge)](https://github.com/kubewarden/community/blob/main/REPOSITORIES.md#sandbox)
+[![License: Apache 2.0](https://img.shields.io/badge/License-Apache2.0-brightgreen.svg)](https://opensource.org/licenses/Apache-2.0)
+[![npm version](https://badge.fury.io/js/kubewarden-policy-sdk.svg)](https://www.npmjs.com/package/kubewarden-policy-sdk)
+
+# Kubewarden Policy SDK for JavaScript/TypeScript
+
+> **Warning:** The SDK is experimental and under active development.
+
+The official JavaScript/TypeScript SDK for writing [Kubewarden](https://kubewarden.io/) policies. This SDK allows you to write Kubernetes admission policies using TypeScript/JavaScript that compile to WebAssembly modules.
+
+## Installation
+
+```bash
+npm install kubewarden-policy-sdk
+```
+
+## Quick Start
+
+### Basic Policy Structure
+
+```typescript
+import { Validation, writeOutput } from 'kubewarden-policy-sdk';
+
+function validate() {
+  // Read the admission request
+  const validationRequest = Validation.readValidationRequest();
+  const settings = validationRequest.settings;
+
+  // Your policy logic here
+  const isValid = yourValidationLogic(validationRequest.request);
+
+  // Create response
+  const response = new Validation.ValidationResponse(
+    isValid,
+    isValid ? undefined : 'Request rejected by policy',
+    undefined, // mutated_object (for mutating policies)
+    undefined, // warnings
+    { customData: 'example' }, // annotations
+  );
+
+  // Write the response
+  writeOutput(response);
+}
+
+// Export the validate function
+(globalThis as any).validate = validate;
+```
+
+### Using Host Capabilities
+
+> [!IMPORTANT]  
+> Logging to `stdout` or `stderr` will break your policy. Always use `console.error()` for logging instead of `console.log()` to avoid policy failures.
```

Comment:
```suggestion
> Logging to `stdout` will break your policy. Always use `console.error()` for logging instead of `console.log()` to avoid policy failures.
```

Printing to stderr (what `console.error()` does) is fine. We [emit warnings](https://github.com/kubewarden/policy-evaluator/blob/main/src/runtimes/wasi_cli/runtime.rs#L40-L47) for these.